src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProjectProvider.cs

```diff
@@ -74,37 +74,31 @@ public NetCorePackageReferenceProjectProvider(IProjectSystemCache projectSystemC
                 return null;
             }
 
-            // Check if the project is not CPS capable or if it is CPS capable then it does not have TargetFramework(s), if so then return false
-            if (!hierarchy.IsCapabilityMatch("CPS"))
+            // Check if the project is not CPS capable or if it is CPS capable that its opt'd in PackageReferences
+            if (!VsHierarchyUtility.IsCPSCapabilityCompliant(hierarchy) ||
```

Comment:
So I was doing some related work and now I'm wondering if this is potentially too relaxed. 

This now checks for "CPS" or `VisualC`.
Not sure which projects are `VisualC`, any reference where I can learn about that? Will projects that declare VisualC go through the nomination workflow? 

Next q is are all these projects guaranteed to have UnconfiguredProject? It might be worth adding a null check for that too, because NuGet wouldn't be able to write to the project file otherwise.

Reply:
I finally found the doc I needed, https://github.com/microsoft/VSProjectSystem/blob/master/doc/overview/project_capabilities.md. 

Does `PackageREferences` as a capability imply nomination + UnconfiguredProject? 

NuGet assumes that right now.

Reply:
I basically ended up running into this while working on https://github.com/NuGet/Home/pull/10064.

Reply:
Disregard the previous versions of comments. Missed the double negative. Updating the first comment with more details